### ISSUE TITLE
parallel calls to 'conda create' corrupt packages and cached repo data

### ISSUE THREAD

**Issue by najtin** [conda/conda#13037]
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

After creating a bunch of environments (possibly in parallel), conda crashes. This is different from #12995 and #13026 because the decoder does not crash a line 0 col 0 but 280 chars into the json. According to `jq` all files at  /home/user/miniconda3/pkgs/cache/ are valid json. 

Steps to reproduce 
```bash
# install nextflow
curl -s https://get.nextflow.io | bash
git clone https://github.com/nf-core/rnaseq
mkdir /tmp/testout
./nextflow run rnaseq/main.nf -profile test,conda --max_memory 6.GB --outdir=/tmp/testout
# wait until it fails during the creation of conda environments
# try to install anything using conda
conda install zsh
```

```
Channels:
 - conda-forge
 - bioconda
 - defaults
Platform: linux-64
Collecting package metadata (repodata.json): failed

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/exception_handler.py", line 17, in __call__
        return func(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/cli/main.py", line 64, in main_subshell
        exit_code = do_call(args, parser)
                    ^^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/cli/conda_argparse.py", line 167, in do_call
        result = getattr(module, func_name)(args, parser)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/notices/core.py", line 124, in wrapper
        return func(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/cli/main_install.py", line 26, in execute
        install(args, parser, "install")
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/cli/install.py", line 316, in install
        unlink_link_transaction = solver.solve_for_transaction(
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/core/solve.py", line 154, in solve_for_transaction
        unlink_precs, link_precs = self.solve_for_diff(
                                   ^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/core/solve.py", line 215, in solve_for_diff
        final_precs = self.solve_final_state(
                      ^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda_libmamba_solver/solver.py", line 199, in solve_final_state
        index = IndexHelper(
                ^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda_libmamba_solver/index.py", line 125, in __init__
        self._index = self._load_channels()
                      ^^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda_libmamba_solver/index.py", line 276, in _load_channels
        jsons = {url: str(path) for (url, path) in executor.map(self._fetch_channel, urls)}
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda_libmamba_solver/index.py", line 276, in <dictcomp>
        jsons = {url: str(path) for (url, path) in executor.map(self._fetch_channel, urls)}
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/concurrent/futures/_base.py", line 619, in result_iterator
        yield _result_or_cancel(fs.pop())
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/concurrent/futures/_base.py", line 317, in _result_or_cancel
        return fut.result(timeout)
               ^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/concurrent/futures/_base.py", line 449, in result
        return self.__get_result()
               ^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
        raise self._exception
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/concurrent/futures/thread.py", line 58, in run
        result = self.fn(*self.args, **self.kwargs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda_libmamba_solver/index.py", line 231, in _fetch_channel
        json_path, _ = subdir_data.repo_fetch.fetch_latest_path()
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/gateways/repodata/__init__.py", line 770, in fetch_latest_path
        _, state = self.fetch_latest()
                   ^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/gateways/repodata/__init__.py", line 820, in fetch_latest
        cache.load_state()
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/gateways/repodata/__init__.py", line 626, in load_state
        self.load(state_only=True)
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/gateways/repodata/__init__.py", line 576, in load
        state = json.loads(state_file.read())
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/json/__init__.py", line 346, in loads
        return _default_decoder.decode(s)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/json/decoder.py", line 340, in decode
        raise JSONDecodeError("Extra data", s, end)
    json.decoder.JSONDecodeError: Extra data: line 9 column 2 (char 280)

`$ /home/rnaseq-analyzer/miniconda3/bin/conda install zsh`

  environment variables:
                 CIO_TEST=<not set>
        CONDA_DEFAULT_ENV=base
                CONDA_EXE=/home/rnaseq-analyzer/miniconda3/bin/conda
             CONDA_PREFIX=/home/rnaseq-analyzer/miniconda3
    CONDA_PROMPT_MODIFIER=(base)
         CONDA_PYTHON_EXE=/home/rnaseq-analyzer/miniconda3/bin/python
               CONDA_ROOT=/home/rnaseq-analyzer/miniconda3
              CONDA_SHLVL=1
           CURL_CA_BUNDLE=<not set>
               LD_PRELOAD=<not set>
                     PATH=/home/rnaseq-analyzer/miniconda3/bin:/home/rnaseq-
                          analyzer/miniconda3/condabin:/home/rnaseq-analyzer/miniconda3/bin:/usr
                          /local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/rnaseq-
                          analyzer/.local/bin
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>
```

### Conda Info

```shell
active environment : base
    active env location : /home/rnaseq-analyzer/miniconda3
            shell level : 1
       user config file : /home/rnaseq-analyzer/.condarc
 populated config files : /home/rnaseq-analyzer/.condarc
          conda version : 23.7.3
    conda-build version : not installed
         python version : 3.11.4.final.0
       virtual packages : __archspec=1=x86_64
                          __glibc=2.35=0
                          __linux=5.15.0=0
                          __unix=0=0
       base environment : /home/rnaseq-analyzer/miniconda3  (writable)
      conda av data dir : /home/rnaseq-analyzer/miniconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/bioconda/linux-64
                          https://conda.anaconda.org/bioconda/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/rnaseq-analyzer/miniconda3/pkgs
                          /home/rnaseq-analyzer/.conda/pkgs
       envs directories : /home/rnaseq-analyzer/miniconda3/envs
                          /home/rnaseq-analyzer/.conda/envs
               platform : linux-64
             user-agent : conda/23.7.3 requests/2.31.0 CPython/3.11.4 Linux/5.15.0-78-generic ubuntu/22.04.3 glibc/2.35 solver/libmamba conda-libmamba-solver/23.7.0 libmambapy/1.4.1
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```

### Conda Config

```shell
==> /home/rnaseq-analyzer/.condarc <==
channels:
  - conda-forge
  - bioconda
  - defaults
solver: libmamba

==> cmd_line <==
debug: False
json: False
```

### Conda list

```shell
# packages in environment at /home/rnaseq-analyzer/miniconda3:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                  2_kmp_llvm    conda-forge
boltons                   23.0.0          py311h06a4308_0    defaults
brotlipy                  0.7.0           py311h5eee18b_1002    defaults
bzip2                     1.0.8                h7b6447c_0    defaults
c-ares                    1.19.0               h5eee18b_0    defaults
ca-certificates           2023.7.22            hbcca054_0    conda-forge
certifi                   2023.7.22          pyhd8ed1ab_0    conda-forge
cffi                      1.15.1          py311h5eee18b_3    defaults
charset-normalizer        2.0.4              pyhd3eb1b0_0    defaults
conda                     23.7.2          py311h38be061_0    conda-forge
conda-content-trust       0.1.3           py311h06a4308_0    defaults
conda-libmamba-solver     23.7.0             pyhd8ed1ab_0    conda-forge
conda-package-handling    2.2.0           py311h06a4308_0    defaults
conda-package-streaming   0.9.0           py311h06a4308_0    defaults
cryptography              39.0.1          py311h9ce1e76_2    defaults
fmt                       9.1.0                hdb19cb5_0    defaults
htslib                    1.17                 h81da01d_2    bioconda
icu                       58.2                 he6710b0_3    defaults
idna                      3.4             py311h06a4308_0    defaults
jsonpatch                 1.32               pyhd3eb1b0_0    defaults
jsonpointer               2.1                pyhd3eb1b0_0    defaults
krb5                      1.20.1               h143b758_1    defaults
ld_impl_linux-64          2.38                 h1181459_1    defaults
libarchive                3.6.2                h6ac8c49_2    defaults
libcurl                   8.1.1                h251f7ec_2    defaults
libdeflate                1.18                 h0b41bf4_0    conda-forge
libedit                   3.1.20221030         h5eee18b_0    defaults
libev                     4.33                 h7f8727e_1    defaults
libffi                    3.4.4                h6a678d5_0    defaults
libgcc-ng                 13.1.0               he5830b7_0    conda-forge
libmamba                  1.4.1                h2dafd23_1    defaults
libmambapy                1.4.1           py311h2dafd23_1    defaults
libnghttp2                1.52.0               h2d74bed_1    defaults
libsolv                   0.7.22               he621ea3_0    defaults
libssh2                   1.10.0               hdbd6064_2    defaults
libstdcxx-ng              11.2.0               h1234567_1    defaults
libuuid                   1.41.5               h5eee18b_0    defaults
libxml2                   2.10.4               hcbfbd50_0    defaults
libzlib                   1.2.13               hd590300_5    conda-forge
llvm-openmp               16.0.6               h4dfa4b3_0    conda-forge
lz4-c                     1.9.4                h6a678d5_0    defaults
ncurses                   6.4                  h6a678d5_0    defaults
openssl                   3.1.2                hd590300_0    conda-forge
packaging                 23.0            py311h06a4308_0    defaults
pcre2                     10.37                he7ceb23_1    defaults
pip                       23.1.2          py311h06a4308_0    defaults
pluggy                    1.0.0           py311h06a4308_1    defaults
pybind11-abi              4                    hd3eb1b0_1    defaults
pycosat                   0.6.4           py311h5eee18b_0    defaults
pycparser                 2.21               pyhd3eb1b0_0    defaults
pyopenssl                 23.2.0          py311h06a4308_0    defaults
pysocks                   1.7.1           py311h06a4308_0    defaults
python                    3.11.4               h955ad1f_0    defaults
python_abi                3.11                    2_cp311    conda-forge
readline                  8.2                  h5eee18b_0    defaults
reproc                    14.2.4               h295c915_1    defaults
reproc-cpp                14.2.4               h295c915_1    defaults
requests                  2.31.0          py311h06a4308_0    defaults
ruamel.yaml               0.17.21         py311h5eee18b_0    defaults
samtools                  1.17                 hd87286a_1    bioconda
setuptools                67.8.0          py311h06a4308_0    defaults
six                       1.16.0             pyhd3eb1b0_1    defaults
sqlite                    3.41.2               h5eee18b_0    defaults
tk                        8.6.12               h1ccaba5_0    defaults
toolz                     0.12.0          py311h06a4308_0    defaults
tqdm                      4.65.0          py311h92b7b1e_0    defaults
tzdata                    2023c                h04d1e81_0    defaults
urllib3                   1.26.16         py311h06a4308_0    defaults
wheel                     0.38.4          py311h06a4308_0    defaults
xz                        5.4.2                h5eee18b_0    defaults
yaml-cpp                  0.7.0                h295c915_1    defaults
zlib                      1.2.13               hd590300_5    conda-forge
zstandard                 0.19.0          py311h5eee18b_0    defaults
zstd                      1.5.5                hc292b87_0    defaults
```

**Comment by najtin** [comment]
`conda clean --all` does *temporarily* resolve the issue. The problem is that nextflow generates many, many environments back to back. When the creation of one environment fails, then the whole pipelines fails. This totally breaks nextflow. Specifically all our automated process stop working. It is not really an option to restart the process every time this happens, because all processing steps done up to this point would also have to be repeated. 

Basically, nextflow breaks if conda corrupts the cache files so often by itself. 

On main, there is even a line in https://github.com/conda/conda/blob/36fb7265ca75008e166ec0b3dfef30fdfacd8342/conda/gateways/repodata/__init__.py#L582 where the author asks if we should catch the decode error as well there. But i can not yet confirm or deny if this would fix the issue.

**Comment by najtin** [comment]
Okay, simply adding JSONDecodeError to the exect does not resolve the issue.

**Comment by najtin** [comment]
How can i forcibly reload the repodata if the decode fails to handle this gracefully?

**Comment by najtin** [comment]
I was able to implement the changes necessary to force a reload upon a JSONDecodeError https://github.com/najtin/conda/commit/70356b8b2a2d953c958b4341c8f43918284ee2d1 This fixes the cache issue. 

However, now i get other corruption related errors, for example:
```
ERROR ~ Error executing process > 'NFCORE_RNASEQ:RNASEQ:FASTQ_FASTQC_UMITOOLS_TRIMGALORE:FASTQC (3)'

Caused by:
  Failed to create Conda environment
  command: conda create --mkdir --yes --quiet --prefix /home/rnaseq-analyzer/work/conda/env-e92705f017a3fab1fca3da28471361dd bioconda::fastqc=0.11.9
  status : 1
  message:
    CRITICAL concurrent.futures:_worker(109): Exception in worker
    Traceback (most recent call last):
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/core/path_actions.py", line 1391, in execute
        raw_index_json = read_index_json(self.target_full_path)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/gateways/disk/read.py", line 130, in read_index_json
        with open(join(extracted_package_directory, "info", "index.json")) as fi:
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/common/compat.py", line 71, in open
        return io_open(
               ^^^^^^^^
    FileNotFoundError: [Errno 2] No such file or directory: '/home/rnaseq-analyzer/miniconda3/pkgs/perl-5.32.1-4_hd590300_perl5/info/index.json'
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/concurrent/futures/thread.py", line 83, in _worker
        work_item.run()
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/concurrent/futures/thread.py", line 60, in run
        self.future.set_exception(exc)
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/concurrent/futures/_base.py", line 565, in set_exception
        self._invoke_callbacks()
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/concurrent/futures/_base.py", line 340, in _invoke_callbacks
        callback(self)
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/core/package_cache_data.py", line 941, in done_callback
        future.result()
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/concurrent/futures/_base.py", line 449, in result
        return self.__get_result()
               ^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
        raise self._exception
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/concurrent/futures/thread.py", line 58, in run
        result = self.fn(*self.args, **self.kwargs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/core/package_cache_data.py", line 916, in do_extract_action
        extract_action.execute(None)
      File "/home/rnaseq-analyzer/miniconda3/lib/python3.11/site-packages/conda/core/path_actions.py", line 1400, in execute
        sys.exit(1)
    SystemExit: 1
    
    InvalidArchiveError('Error with archive /home/rnaseq-analyzer/miniconda3/pkgs/icu-73.2-h59595ed_0.conda.  You probably need to delete and re-download or re-create this file.  Message was:\n\nfailed with error: File is not a zip file')
    InvalidArchiveError('Error with archive /home/rnaseq-analyzer/miniconda3/pkgs/openjdk-20.0.0-h8e330f5_0.conda.  You probably need to delete and re-download or re-create this file.  Message was:\n\nfailed with error: File is not a zip file')



 -- Check '.nextflow.log' file for details
```


We can reproduce this on different machines. I think this is a race condition. nextflow starts many conda processes in parallel.

**Comment by najtin** [comment]
Okay, i'm pretty sure that this is some race condition within conda. I modified the conda executable so that only one conda process runs at a time. This completely  fixes all corruption issues (also the corrupted repo data). I have to release the lock manually from time to time though. 
```python
!/home/rnaseq-analyzer/miniconda3/bin/python
# -*- coding: utf-8 -*-
import sys
import posix_ipc
# Before any more imports, leave cwd out of sys.path for internal 'conda shell.*' commands.
# see https://github.com/conda/conda/issues/6549
if len(sys.argv) > 1 and sys.argv[1].startswith('shell.') and sys.path and sys.path[0] == '':
    # The standard first entry in sys.path is an empty string,
    # and os.path.abspath('') expands to os.getcwd().
    del sys.path[0]

if __name__ == '__main__':
    try:
        semaphore = posix_ipc.Semaphore("conda-main-semaphore", posix_ipc.O_CREX)
    except posix_ipc.ExistentialError:
        semaphore = posix_ipc.Semaphore("conda-main-semaphore")
    error = None
    from conda.cli import main
    try:
        semaphore.acquire()
        exit_code = main()
    except Exception as e:
        error = e
    finally:
        semaphore.release()
        semaphore.close()
    if error is not None:
        raise error
    sys.exit(exit_code)
```

This means, that the parallel execution of conda corrupts package, and cache data.

**Comment by najtin** [comment]
I added a ways to cope with recursive conda calls inside of conda itself (sometimes conda info is called) and made it cross platform compatible. Should i open a PR for this?

```python
#!/home/rnaseq-analyzer/miniconda3/bin/python
# -*- coding: utf-8 -*-
import os
import psutil
from filelock import FileLock
import tempfile

def process_exists(pid):
    try:
        process = psutil.Process(pid)
        return True
    except psutil.NoSuchProcess:
        return False

def get_parent_pids():
    pid = os.getpid()
    parent_pids = []
    try:
        process = psutil.Process(pid)
        while process:
            parent_pids.append(process.pid)
            process = process.parent()
    except psutil.NoSuchProcess:
        pass
    
    return parent_pids

import sys

# Before any more imports, leave cwd out of sys.path for internal 'conda shell.*' commands.
# see https://github.com/conda/conda/issues/6549
if len(sys.argv) > 1 and sys.argv[1].startswith('shell.') and sys.path and sys.path[0] == '':
    # The standard first entry in sys.path is an empty string,
    # and os.path.abspath('') expands to os.getcwd().
    del sys.path[0]

conda_lock = tempfile.gettempdir() + "/.conda.pid.lock"
conda_pid = tempfile.gettempdir() + "/.conda.pid"

if __name__ == '__main__':
    use_lock = True
    if os.path.exists(conda_pid):
        with open(conda_pid, "r") as f:
            other_conda_pid = int(f.read())
        if process_exists(other_conda_pid) and other_conda_pid in get_parent_pids():
            use_lock = False
    error = None
    from conda.cli import main
    if use_lock:
        with FileLock(conda_lock):
            with open(conda_pid, "w") as f:
                f.write(str(os.getpid()))
            try:
                exit_code = main()
            except Exception as e:
                error = e
        if error is not None:
            raise error
        sys.exit(exit_code)
    else:
        sys.exit(main())
```

**Comment by najtin** [comment]
This would go here, right? https://github.com/conda/conda/blob/main/conda/cli/main.py#L109

**Comment by najtin** [comment]
What puzzles me is that seemingly very few other people faced this problem. Was some locking mechanism removed in the last releases?

**Comment by dholth** [comment]
We should also need to lock package unpacking and downloads for this one

**Comment by dholth** [comment]
https://github.com/conda/conda/pull/12996 addresses the "corrupt index repo data" part of this issue.

**Comment by birdsarah** [comment]
I'm also facing this issue. Let me know if I can provide any information.

**Comment by dholth** [comment]
@birdsarah Were you seeing package issues or just JSONDecodeError?

**Comment by dholth** [comment]
Thanks for the info. Obviously if two conda's are trying to download or extract the same package at the same time, that's a problem. Depending on the timing, the second conda might think "the folder is there!" and try to link an incomplete set of files or they might simply overwrite each other.

**Comment by najtin** [comment]
I think this boils down to most people expecting that conda is safe to execute in parallel like most other package managers.

**Comment by dholth** [comment]
A long time ago conda had locking. I'm interested in putting an improved lock scheme back in. For example https://github.com/conda/conda/pull/12996, https://github.com/conda/conda/issues/13055

**Comment by Gerry-j** [comment]
My problem was slightly different, when I ran conda clean --all and it didn't solve the problem.

>>conda create -n XXX python=3.8

Collecting package metadata (current_repodata.json): failed

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/exceptions.py", line 1124, in __call__
        return func(*args, **kwargs)
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/cli/main.py", line 69, in main_subshell
        exit_code = do_call(args, p)
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/cli/conda_argparse.py", line 91, in do_call
        return getattr(module, func_name)(args, parser)
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/notices/core.py", line 109, in wrapper
        return func(*args, **kwargs)
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/cli/main_create.py", line 41, in execute
        install(args, parser, 'create')
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/cli/install.py", line 264, in install
        unlink_link_transaction = solver.solve_for_transaction(
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/core/solve.py", line 132, in solve_for_transaction
        unlink_precs, link_precs = self.solve_for_diff(update_modifier, deps_modifier,
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/core/solve.py", line 175, in solve_for_diff
        final_precs = self.solve_final_state(update_modifier, deps_modifier, prune, ignore_pinned,
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/core/solve.py", line 280, in solve_final_state
        ssc = self._collect_all_metadata(ssc)
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/common/io.py", line 84, in decorated
        return f(*args, **kwds)
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/core/solve.py", line 447, in _collect_all_metadata
        index, r = self._prepare(prepared_specs)
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/core/solve.py", line 1060, in _prepare
        reduced_index = get_reduced_index(self.prefix, self.channels,
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/core/index.py", line 267, in get_reduced_index
        new_records = SubdirData.query_all(spec, channels=channels, subdirs=subdirs,
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/core/subdir_data.py", line 124, in query_all
        result = tuple(chain.from_iterable(executor.map(subdir_query, channel_urls)))
      File "/media/b405/opt_disk/anaconda/lib/python3.10/concurrent/futures/_base.py", line 621, in result_iterator
        yield _result_or_cancel(fs.pop())
      File "/media/b405/opt_disk/anaconda/lib/python3.10/concurrent/futures/_base.py", line 319, in _result_or_cancel
        return fut.result(timeout)
      File "/media/b405/opt_disk/anaconda/lib/python3.10/concurrent/futures/_base.py", line 458, in result
        return self.__get_result()
      File "/media/b405/opt_disk/anaconda/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
        raise self._exception
      File "/media/b405/opt_disk/anaconda/lib/python3.10/concurrent/futures/thread.py", line 58, in run
        result = self.fn(*self.args, **self.kwargs)
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/core/subdir_data.py", line 113, in <lambda>
        subdir_query = lambda url: tuple(
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/core/subdir_data.py", line 129, in query
        self.load()
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/core/subdir_data.py", line 215, in load
        _internal_state = self._load()
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/core/subdir_data.py", line 359, in _load
        _internal_state = self._process_raw_repodata_str(raw_repodata_str, mod_etag_headers)
      File "/media/b405/opt_disk/anaconda/lib/python3.10/site-packages/conda/core/subdir_data.py", line 456, in _process_raw_repodata_str
        json_obj = json.loads(raw_repodata_str or "{}")
      File "/media/b405/opt_disk/anaconda/lib/python3.10/json/__init__.py", line 346, in loads
        return _default_decoder.decode(s)
      File "/media/b405/opt_disk/anaconda/lib/python3.10/json/decoder.py", line 337, in decode
        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
      File "/media/b405/opt_disk/anaconda/lib/python3.10/json/decoder.py", line 355, in raw_decode
        raise JSONDecodeError("Expecting value", s, err.value) from None
    json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

`$ /media/b405/opt_disk/anaconda/bin/conda create -n lzr_pat python=3.8`

  environment variables:
                 CIO_TEST=<not set>
        CONDA_DEFAULT_ENV=base
                CONDA_EXE=/media/b405/opt_disk/anaconda/bin/conda
             CONDA_PREFIX=/media/b405/opt_disk/anaconda
    CONDA_PROMPT_MODIFIER=(base)
         CONDA_PYTHON_EXE=/media/b405/opt_disk/anaconda/bin/python
               CONDA_ROOT=/media/b405/opt_disk/anaconda
              CONDA_SHLVL=1
           CURL_CA_BUNDLE=<not set>
               LD_PRELOAD=<not set>
                     PATH=/media/b405/opt_disk/anaconda/bin:/media/b405/opt_disk/anaconda/condab
                          in:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/g
                          ames:/usr/local/games:/snap/bin
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>

     active environment : base
    active env location : /media/b405/opt_disk/anaconda
            shell level : 1
       user config file : /home/b405/.condarc
 populated config files : /home/b405/.condarc
          conda version : 23.1.0
    conda-build version : 3.23.3
         python version : 3.10.9.final.0
       virtual packages : __archspec=1=x86_64
                          __cuda=12.2=0
                          __glibc=2.31=0
                          __linux=5.15.0=0
                          __unix=0=0
       base environment : /media/b405/opt_disk/anaconda  (writable)
      conda av data dir : /media/b405/opt_disk/anaconda/etc/conda
  conda av metadata url : None
           channel URLs : http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/r/linux-64
                          http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/r/noarch
                          http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main/linux-64
                          http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main/noarch
                          http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free/linux-64
                          http://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free/noarch
          package cache : /media/b405/opt_disk/anaconda/pkgs
                          /home/b405/.conda/pkgs
       envs directories : /media/b405/opt_disk/anaconda/envs
                          /home/b405/.conda/envs
               platform : linux-64
             user-agent : conda/23.1.0 requests/2.28.1 CPython/3.10.9 Linux/5.15.0-117-generic ubuntu/20.04.6 glibc/2.31
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
